Add CloudPrem installation page with docker and docker compose. (#31492)
DataDog/documentation: content/en/cloudprem/configure/reverse_connection.md

---
title: Reverse Connection
further_reading:
- link: "/cloudprem/install/"
  tag: "Documentation"
  text: "CloudPrem Installation Prerequisites"
---

> The reverse connection allows your CloudPrem cluster to initiate the HTTP connection with Datadog using your API keys, without the need to set up a DNS entry and public ingress. This is also useful for environments with strict network policies or when you prefer not to expose CloudPrem publicly.

Reviewer: same comment wrt "prefer not to expose CloudPrem publicly"

> The reverse connection allows your CloudPrem cluster to initiate the HTTP connection with Datadog using your API keys, without the need to set up a DNS entry and public ingress. This is also useful for environments with strict network policies or when you prefer not to expose CloudPrem publicly.

Reviewer 2: ```suggestion
The reverse connection allows your CloudPrem cluster to initiate the HTTP connection with Datadog using your API keys, without needing to set up a DNS entry and public ingress, so you don't have to expose CloudPrem to the internet. This is useful for environments with strict network policies or if you prefer to keep CloudPrem private.
```

To activate the reverse connection, set the following values in your Helm chart:

```yaml
config:
  cloudprem:
    enable_reverse_connection: true
    site: "datadoghq.com" # your site

>     dd_api_key: "${DD_API_KEY}"

Reviewer: same wrt env var from secrets

---
title: Reverse Connection
further_reading:
- link: "/cloudprem/install/"
  tag: "Documentation"
  text: "CloudPrem Installation Prerequisites"
---

{{< callout btn_hidden="true" >}}
  Datadog CloudPrem is in Preview.
{{< /callout >}}

## Overview

> Reverse connection lets your CloudPrem cluster initiate an HTTP connection with Datadog using your API key, without requiring adding a DNS entry and public ingress. This setup is useful for environments with strict network policies which do not allows inbound requests.

Reviewer 3: ```suggestion
Reverse connection lets your CloudPrem cluster initiate an HTTP connection with Datadog using your API key, without requiring adding a DNS entry and public ingress. This setup is useful for environments with strict network policies which do not allow inbound requests.
```

> It the default setup for CloudPrem, it requires the following environment variables to be set:

Reviewer 3: Where are these environment variables set?

```suggestion
## Prerequisites

The following environment variables must be set:
```

- `DD_API_KEY`
- `DD_SITE` which defaults to `datadog.com`

## Helm chart configuration

First create a secret to store the Datadog API key:
```
kubectl create secret generic datadog-secret --from-literal DD_API_KEY=<DATADOG_API_KEY>
```

To set

```yaml
environment:
  DD_SITE: <DD_SITE>
  secretRef:
      name: datadog-secret
```

## Proxies

You can configure CloudPrem to use a forward proxy, such as Squid, with the environment variable `HTTPS_PROXY`.

```yaml
environment:
  HTTP_PROXY: <your HTTP proxy>

> 

Reviewer 3: ```suggestion
```
